### PR TITLE
Issue #8473: Records Support check update for RequireThisCheck

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8473: Records Support check update for RequireThisCheck

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/e3988f8092d919b5cbe70f5359c4d9e8/raw/a0f6f5860f8025c19868061dee6e0e40960b992f/projects-to-test-on.properties

Diff Regression config: https://gist.githubusercontent.com/nmancus1/c68d6dc8f36ee2668649b7ce629a6fdb/raw/208d9609dc79c8532f27ba03444cd3f73379e7b7/RequireThisCheck.xml

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordAsTopLevel.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+/* Config:
+ * checkFields = false
+ * checkMethods = true
+ * validateOnlyOverlapping = false
+ *
+ */
+public record InputRequireThisRecordAsTopLevel(int x, int y) {
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordAsTopLevel.java:10]: Please add some usages of the record components (`x` or `y`) to the compact ctor and any method.
All record componens should be passed to `addInstanceMember` method, but do not see such changes in this commit.

Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordAsTopLevel.java:10]: Good catch, I have fixed this and added some more test inputs.

**Comment by nmancus1** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java [hunk 2]
```diff
@@ -130,6 +130,13 @@
  *     return field += "suffix"; // violation: Reference to instance variable "field" needs "this".
  *   }
  * }
+ *
+ * public static record C(int x) {
+ *     public C (int x){
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java:135]: `(int x){`
no space between braces

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java [hunk 3]
```diff
@@ -130,6 +130,13 @@
...
+ *     // Overlapping by constructor argument.
+ *     x = x; // violation: Reference to instance variable "x" needs "this".
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java:137]: 1. indentation is wrong.

2. To be honest, such example with record and constructor seems confusing. May be change constructor to method?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java [hunk 4]
```diff
@@ -210,6 +217,12 @@
  *     s2 += s2; // Violation. Overlapping. Reference to instance variable 's2' needs "this.".
  *   }
  * }
+ *
+ * public static record D(int x) {
+ *     public D{
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java:222]: `D{` space is needed

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java (hunk 2 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Fixed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java (hunk 3 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Fixed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java (hunk 4 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Fixed.

**Pull request merged**